pkg/discovery/network/generic_test.go

```diff
@@ -17,54 +17,93 @@ limitations under the License.
 package network
 
 import (
+	"fmt"
+
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/kubernetes/fake"
+	fakecorev1 "k8s.io/client-go/kubernetes/typed/core/v1/fake"
+	"k8s.io/client-go/testing"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
 
 const testPodCIDR = "1.2.3.4/16"
 const testServiceCIDR = "4.5.6.7/16"
+const testServiceCIDRFromService = "7.8.9.10/16"
 
 var _ = Describe("discoverGenericNetwork", func() {
-	When("There are no generic k8s pods to look at", func() {
-		It("Should return nil cluster network", func() {
-			clusterNet := testDiscoverGenericWith()
-			Expect(clusterNet).To(BeNil())
-		})
-	})
-
 	When("There is a kube-proxy with no expected parameters", func() {
-		It("Should return nil cluster network", func() {
-			clusterNet := testDiscoverGenericWith(
+		var clusterNet *ClusterNetwork
+
+		BeforeEach(func() {
+			clusterNet = testDiscoverGenericWith(
 				fakePod("kube-proxy", []string{"kube-proxy", "--cluster-ABCD=1.2.3.4"}, []v1.EnvVar{}),
 			)
-			Expect(clusterNet).To(BeNil())
+			Expect(clusterNet).NotTo(BeNil())
+		})
+
+		It("Should return the ClusterNetwork structure with empty PodCIDRs", func() {
+			Expect(clusterNet.PodCIDRs).To(BeEmpty())
+		})
+
+		It("Should identify the networkplugin as generic", func() {
+			Expect(clusterNet.NetworkPlugin).To(BeIdenticalTo("generic"))
+		})
+
+		It("Should return the ClusterNetwork structure with the service CIDR", func() {
+			Expect(clusterNet.ServiceCIDRs).To(Equal([]string{testServiceCIDRFromService}))
 		})
 	})
 
 	When("There is a kube-controller with no expected parameters", func() {
-		It("Should return nil cluster network", func() {
-			clusterNet := testDiscoverGenericWith(
+		var clusterNet *ClusterNetwork
+
+		BeforeEach(func() {
+			clusterNet = testDiscoverGenericWith(
 				fakePod("kube-controller", []string{"kube-controller", "--cluster-ABCD=1.2.3.4"}, []v1.EnvVar{}),
```

Comment:
```suggestion
				fakePod("kube-controller-manager", []string{"kube-controller-manager", "--cluster-ABCD=1.2.3.4"}, []v1.EnvVar{}),
```

Reply:
In the code, we are actually checking for component `kube-controller-manager` and this test-case is about validating the arguments (i.e., no expected params).

Reply:
Thank you for pointing it out. Fixed.